keep/providers/aks_provider/aks_provider.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+import os
+
+import pydantic
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from azure.identity import DefaultAzureCredential
+from azure.mgmt.containerservice import ContainerServiceClient
+
+from keep.contextmanager.contextmanager import ContextManager
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+from keep.providers.providers_factory import ProvidersFactory
+
+@pydantic.dataclasses.dataclass
+class AzureProviderAuthConfig:
+    """Azure AKS authentication configuration."""
+
+    resource_group: str = pydantic.Field(
+        metadata={"required": True, "description": "Azure Resource Group name"}
+    )
+    cluster_name: str = pydantic.Field(
+        metadata={"required": True, "description": "Azure AKS cluster name"}
+    )
+
+class AzureProvider(BaseProvider):
+    def __init__(
+        self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        super().__init__(context_manager, provider_id, config)
+        self._resource_group = self.authentication_config.resource_group
+        self._cluster_name = self.authentication_config.cluster_name
+        self._kubeconfig_path = "/path/to/azure/kubeconfig.yaml"
```

Comment:
this is not acceptable since the customer doesn't upload `kubeconfig.yaml`, the authentication part is the main issue on this provider - a customer should be able to connect its AKS from the UI

Reply:
Ah alright, thinking...